modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java

```diff
@@ -41,26 +40,30 @@ public class MainResponse extends ActionResponse implements ToXContentObject {
         super(in);
         nodeName = in.readString();
         version = Version.readVersion(in);
-        indexVersion = in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_031) ? IndexVersion.readVersion(in) : null;
-        transportVersion = in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_019) ? TransportVersion.readVersion(in) : null;
+
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_034)) {
+            luceneVersion = in.readString();
+        } else {
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_031)) {
```

Comment:
Worth a few comments here explaining what's going on (and why we're reading values then throwing them away)

Reply:
Added in https://github.com/elastic/elasticsearch/pull/97675/commits/5ed9d69842d02099f26241b0d7215343151b8a5a